Minor release: 0.15.1

scheibelp opened the pull request.

michaelkuhn committed:
autotools: Fix config.guess detection, take two (#17333)

The previous fix from #17149 contained a thinko that produced errors for
packages that overwrite configure_directory.

adamjstewart committed:
Fix Intel MPI super invocation, again (#17378)

tz-rrze committed:
inel-mpi: fix for wrong structure name instroduced in ea8a0be4 (#17382)

it's    mpi_compiler_wrappers
and not mpi_compiler._wrappers

fixes 2nd part of #17371

tgamblin committed:
bugfix: no infinite recursion in setup-env.sh on Cray

On Cray platforms, we rely heavily on the module system to figure out
what targets, compilers, etc. are available. This unfortunately means
that we shell out to the `module` command as part of platform
initialization.

Because we run subcommands in a shell, we can get infinite recursion if
`setup-env.sh` and friends are in some init script like `.bashrc`.

This fixes the infinite loop by adding guards around `setup-env.sh`,
`setup-env.csh`, and `setup-env.fish`, to prevent recursive
initializations of Spack. This is safe because Spack never shells out to
itself, so we do not need it to be initialized in subshells.

- [x] add recursion guard around `setup-env.sh`
- [x] add recursion guard around `setup-env.csh`
- [x] add recursion guard around `setup-env.fish`

michaelkuhn committed:
autotools bugfix: handle missing config.guess (#17356)

Spack was attempting to calculate abspath on the located config.guess
path even when it was not found (None); this commit skips the abspath
calculation when config.guess is not found.

cedricchevalier19 committed:
Fix gcc + binutils compilation. (#9024)

* fix binutils deptype for gcc

binutils needs to be a run dependency of gcc

* Fix gcc+binutils build on RHEL7+

static-libstdc++ is not available with system gcc.
Anyway, as it is for bootstraping, we do not really care depending on
a shared libstdc++.

Co-authored-by: Michael Kuhn <michael@ikkoku.de>

scheibelp committed:
add public spack mirror (#17077)

becker33 committed:
installation: skip repository metadata for externals (#16954)

When Spack installs a package, it stores repository package.py files
for it and all of its dependencies - any package with a Spack metadata
directory in its installation prefix.

It turns out this was too broad: this ends up including external
packages installed by Spack (e.g. installed by another Spack instance).
Currently Spack doesn't store the namespace properly for such packages,
so even though the package file could be fetched from the external,
Spack is unable to locate it.

This commit avoids the issue by skipping any attempt to locate and copy
from the package repository of externals, regardless of whether they
have a Spack repo directory.

scheibelp committed:
Bump version to 0.15.2; update CHANGELOG and version references

mwkrentel commented:
@scheibelp Is commit #17356 (autotools take three) going to be in 0.15.1?
Without it, the autotools fix in #17333 (autotools take two) simply fixes one
bug but creates another.

Right now, hpctoolkit is broken on ppc/arm in 1.15.0 due to a late
commit before the tag.  It's fixed in develop, but without #17356, it will remain
broken in 1.15.1, just broken in a different way.

Edit: Ok, nevermind, I see that it's there.  The subject line used to
say, 'take three' but it changed and I missed it.

becker33 committed:
spack install: improve error message with no args (#17454)

The error message was not updated when the behavior of Spack environments
was changed to not automatically activate the local environment in #17258.
The previous error message no longer makes sense.
